trafficlight/tests/dehydrated_device_testsuite.py

```diff
@@ -0,0 +1,206 @@
+                            "data": login_data_alice,
+                            "responses": {"registered": "register_b"},
+                        },
+                    },
+                ),
+                ModelState(
+                    "register_b",
+                    {
+                        bob: {
+                            "action": "register",
+                            "data": login_data_bob,
+                            "responses": {"registered": "enable_dehydrated_device"},
+                        }
+                    },
+                ),
+                # Enabling dehydrated devices in element-web does not work as expected.
+                # The dehydrated device is only uploaded when you do a second login and restore
+                # from key backup. So this is a failing test for now.
+                ModelState(
+                    "enable_dehydrated_device",
+                    {
+                        alice: {
+                            "action": "enable_dehydrated_device",
+                            "data": {
+                                "key_backup_passphrase": "helloworld123helloworld"
+                            },
+                            "responses": {"enabled_dehydrated_device": "create_room"},
+                        }
+                    },
+                ),
+                ModelState(
+                    "create_room",
+                    {
+                        bob: {
+                            "action": "create_room",
+                            "data": {"name": "little test room"},
+                            "responses": {"room_created": "invite_user"},
+                        }
+                    },
+                ),
+                ModelState(
+                    "invite_user",
+                    {
+                        bob: {
+                            "action": "invite_user",
+                            "data": {
+                                "userId": f'{login_data_alice["username"]}:{servers[0].server_name}'
+                            },
+                            "responses": {"invited": "accept_invite"},
+                        }
+                    },
+                ),
+                ModelState(
+                    "accept_invite",
+                    {
+                        alice: {
+                            "action": "accept_invite",
+                            "responses": {"accepted": "alice_logs_out"},
+                        }
+                    },
+                ),
+                ModelState(
+                    "alice_logs_out",
+                    {
+                        alice: {
+                            "action": "logout",
+                            "data": {},
+                            "responses": {"logged_out": "bob_advances_time"},
+                        }
+                    },
+                ),
+                ModelState(
+                    "bob_advances_time",
```

Comment:
Is this a valid thing to do? i know it'll test the system, but is bob not now out of sync with alice in terms of local clocks?

Reply:
(this isn't the same as waiting, as that puts both clocks forward

Reply:
This is for step 6 as listed at https://github.com/vector-im/hydrogen-internal/issues/21. This would cause the clocks to be out of sync but does it have any real consequence?